applications/ray/kuberay-tpu-webhook/webhook_main_test.go

```diff
@@ -0,0 +1,595 @@
+package main
+
+import (
+	"testing"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+
+	"github.com/stretchr/testify/assert"
+	// "github.com/GoogleCloudPlatform/kuberay-tpu-webhook"
+)
+
+var (
+	namespaceStr                  string
+	instanceName                  string
+	groupNameStr                  string
+	headGroupNameStr              string
+	testPodAdmissionReviews       *admissionv1.AdmissionReview
+	testCPUPods                   []runtime.Object
+	testTPUPods                   []runtime.Object
+	testRayClusterAdmissionReview *admissionv1.AdmissionReview
+	testRayClusterNoTPUs          *rayv1.RayCluster
+	testRayClusterSingleHostTPU   *rayv1.RayCluster
+	testRayClusterMultiHostTPU    *rayv1.RayCluster
+	testServices                  []runtime.Object
+	workerSelector                labels.Selector
+	headNodeIP                    string
+	// sliceToWorkers				  map[slice][]worker
```

Comment:
Is this intended to use the same copy of `sliceToWorkers` in the webhook?

Reply:
Yeah it uses the same `sliceToWorkers` since this is what's added to internally in `getNextWorkerID`. I'll add in some logic to save a deep copy of `sliceToWorkers` and return it to its state after the test. I don't think this should cause any issues unless you run the tests while the webhook is intercepting pods with the same cluster name, namespace, and group name.